tests/test_frompyobject.rs

```diff
@@ -170,6 +186,108 @@ fn test_transparent_tuple_struct() {
     assert_eq!(tup.0, "test");
 }
 
+#[pyclass]
+struct PyBaz {
+    #[pyo3(get)]
+    tup: (String, String),
+    #[pyo3(get)]
+    e: PyE,
+}
+
+#[derive(Debug, FromPyObject)]
+struct Baz<U, T> {
+    e: E<U, T>,
+    tup: Tuple,
+}
+
+#[test]
+fn test_struct_nested_type_errors() {
+    let gil = Python::acquire_gil();
+    let py = gil.python();
```

Comment:
Nit: can we please use `Python::with_gil` rather than `acquire_gil` in this and all the other new tests please? We're intending to move away from `acquire_gil` because of underlying challenges with that API.

Reply:
:+1: